Enhancement: remote Pull Request review

jodosha opened the pull request:
_WIP_

mattzcarey commented:
Is there a reason to use both the GraphQL client and the rest client. Can we not get everything done with the rest client? 

Also you have added in the function to get files in the review command. This should be clean of git commands or api calls and just focus on the review logic.

If you look at the roadmap this will become clear. We are building a backend which will handle a lot of the functionality you want.

jodosha commented:
@mattzcarey Hey, thanks for the review. 

REST vs GraphQL: I checked, it’s possible. The code of this PR is a simplified version of a larger program that required GQL. I will change this one.

Args of review: will change this as well.

backend vs script: For my use case I will need to integrate your project into a larger set of data. Having CLI is perfect point of integration. It’s UNIX-like approach. A backend has the risk of trapping the data. Does this make sense? 🙂

I’m on vacation ATM, I will make the changes in a couple of weeks, hoping you will have accept this enhancement. 🙂

jodosha commented:
@mattzcarey 
> Hey, thanks for your message and I hope you have a great holiday!
Thank you, I had!

---

* ✅  I moved the retrieval of files outside of `review` (see https://github.com/mattzcarey/code-review-gpt/pull/96/commits/1b4b0fba2c791954871f9250db059c8876335a2e)
* ✅ I simplified **a lot** the code by using GitHub REST API only. (see https://github.com/mattzcarey/code-review-gpt/pull/96/commits/0150edd1cee9cffc1bd9cc328472e3339d7cfb48)

mattzcarey commented:
Looks great. Will functionally review now. :)

mattzcarey commented:
I am getting this error: 

```
ERROR   Error: Error: Failed to get remote Pull Request files: Error: fetch is not set. Please pass a fetch implementation as new Octokit({ request: { fetch }}). Learn more at https://github.com/octokit/octokit.js/#fetch-missing
```

jodosha commented:
@mattzcarey Can you please tell me step by step how you manual test it? E.g. do you transpile from TS? So that I can reproduce your problem.

mattzcarey commented:
I pulled your branch and ran 
```
npm start -- --remote=mattzcarey/code-review-gpt#96 --debug=true
```

jodosha commented:
@mattzcarey Which Node version are you using?

Newer versions of Octokit (GitHub SDK) require Node.js 18+.
See https://github.com/octokit/octokit.js/#fetch-missing

I suggest enforcing the project to support Node 18+.
Because 16 is EOL and 18 is LTS.

jodosha commented:
@mattzcarey I can't reproduce the problem. This is my output: https://gist.github.com/jodosha/8f84c425389ea5d3a182852c8817975b

mattzcarey commented:
Should be good. I'll add that to the project.

mattzcarey commented:
Thanks for your contribution! @jodosha

jodosha commented:
@mattzcarey Can you please consider releasing a new version so that I can use the tool as a globally installed CLI? Thanks. 🙂

mattzcarey commented:
> @mattzcarey Can you please consider releasing a new version so that I can use the tool as a globally installed CLI? Thanks. 🙂

Yes it will be done this week. :)
